Gems/OpenXRVk/Code/Source/InputDeviceXRController.cpp

```diff
@@ -419,4 +450,307 @@ namespace AzFramework
+            if (const AZ::RPI::ViewportContextPtr viewportContext = viewportContextMgr->GetDefaultViewportContext();
+                viewportContext != nullptr)
+            {
+                if (const AZ::RPI::ViewPtr view = viewportContext->GetDefaultView();
+                    view != nullptr)
+                {
+                    return view->GetCameraTransform();
+                }
+            }
+        }
+        return AZ::Transform::CreateIdentity();
+    }
+
+    static void DrawControllerAxes(DebugDisplayRequests& debugDisplay, const AZ::Vector3& position, const AZ::Quaternion& orientation)
+    {
+        static const AZ::Color axisColorX(1.f, 0.f, 0.f, 0.9f);
+        static const AZ::Color axisColorY(0.f, 1.f, 0.f, 0.9f);
+        static const AZ::Color axisColorZ(0.f, 0.f, 1.f, 0.9f);
+
+        const auto cameraTransform = GetCameraTransformFromCurrentView();
+        const AZ::Vector3& cameraPosition = cameraTransform.GetTranslation();
+        const AZ::Vector3 controllerPosition = cameraPosition + position;
+
+        const AZ::Transform controllerTransform = AZ::Transform::CreateFromQuaternionAndTranslation(orientation, controllerPosition);
+        debugDisplay.SetColor(axisColorX);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisX());
+        debugDisplay.SetColor(axisColorY);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisY());
+        debugDisplay.SetColor(axisColorZ);
+        debugDisplay.DrawLine(controllerPosition, controllerPosition + controllerTransform.GetBasisZ());
+    }
+
+#endif // !_RELEASE
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceXRController::CheckDebugDrawCheat() const
+    {
+        // This looks for specific controller input and will toggle the debug draw cvar.
+        const auto& rawControllerData = m_impl->GetRawState();
+        using xrc = InputDeviceXRController;
+        static bool cheatWasPressed = false;
+
+        // Menu button + Left Trigger pulled past 0.9 will toggle.
+        // To avoid button bounce, block re-toggle until the menu button is released.
+        const bool menuPressed = rawControllerData.GetDigitalButtonState(xrc::Button::Menu);
+        const float leftTrigger = rawControllerData.GetLeftTriggerAdjustedForDeadZoneAndNormalized();
+        if (menuPressed)
+        {
+            if (!cheatWasPressed && leftTrigger > 0.9f)
+            {
+                cheatWasPressed = true;
+                OpenXRVk::xr_DebugDrawInput = !OpenXRVk::xr_DebugDrawInput;
+            }
+        }
+        else
+        {
+            cheatWasPressed = false;
+        }
+    }
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////
+    void InputDeviceXRController::DrawGlobalDebugInfo()
+    {
+        CheckDebugDrawCheat();
+
+#if !defined(_RELEASE)
```

Comment:
Seem `CheckDebugDrawCheat` is not necessary either in release.

````
#if !defined(_RELEASE)
        CheckDebugDrawCheat();
````

Reply:
I agree.  And other feedback wasn't too keen on having a ton of code wrapped in `!_RELEASE`, so I avoided doing that in the header, and in the cpp I wrapped the bodies of these two functions in `!_RELEASE`.  Hopefully that's a good compromise, at the expense that release has these empty functions to strip out.